### PR TITLE
adding dictionaries by aliases

### PR DESCRIPTION
**Pull request by Dmitriy-Yugay**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,8 +41,9 @@
      | DisconnectOnError	| Session setting for doing an automatic disconnect when an error occurs.	| Y / N	| N |    
      | SessionAlias| session alias for incoming/outgoing th2 messages. | case-sensitive alpha-numeric string | |
      | QueueCapacity | maximum size of the session message queue | integer value | |
-     
-     
+     | zipDictionaries | Defines the way dictionaries are loaded. preferred setting is false for downloading by aliases, true for downloading dictionaries from a zip archive  | true / false | false |
+
```
Comment (cordwelt) [inline README.md:67]: Do we really need to keep ability to load dictionaries from zip-archive?

File: src/main/java/com/exactpro/th2/fix/client/Main.java [hunk 2]
```diff
@@ -115,27 +115,36 @@ public static void main(String[] args) throws Exception {
...
+            try {
+                for (String dictionaryAlias : factory.getDictionaryAliases()) {
+                    InputStream dictionary = factory.loadDictionary(dictionaryAlias);
+                    writeDictionary(temporaryDirectory, dictionary, Path.of(dictionaryAlias));
                 }
-            } catch (IncorrectFixFileNameException | ConfigError e) {
-                throw new Exception("Failed to unzip dictionaries along the path: " + zipEntry.getName(), e);
+            } catch (Exception e) {
+                throw new Exception("Failed to create DataDictionary", e);
             }
```
Comment (cordwelt) [inline src/main/java/com/exactpro/th2/fix/client/Main.java]: I'd suggest wrapping cycle-body into try-catch instead. This way we'll have info about which dictionary we've failed to load

File: src/main/java/com/exactpro/th2/fix/client/Main.java [hunk 3]
```diff
@@ -175,6 +182,16 @@ public static void main(String[] args) throws Exception {
 
     }
 
+    private static void writeDictionary(Path directory, InputStream dictionary, Path fileName) throws IOException, ConfigError {
+        Path pathToDictionary = Files.createFile(getPathToDictionary(directory, fileName));
+        Files.write(pathToDictionary, dictionary.readAllBytes());
+        try {
+            new DataDictionary(pathToDictionary.toString()); //check that xml file contains the correct values
+        } catch (ConfigError error) {
+            throw new ConfigError("Failed to create DataDictionary along the path " + fileName.getFileName(), error);
```
Comment (cordwelt) [inline src/main/java/com/exactpro/th2/fix/client/Main.java:191]: > new ConfigError

Not sure that we should use QFJ's exceptions for our purposes, but it's probably ok

File: README.md (hunk 1 above)
Comment (Dmitriy-Yugay) [reply README.md:67]: yep, for backward compatibility purposes
not everywhere has the necessary infra yet

**Review by cordwelt — APPROVED** [review]

**Pull request merged**